modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1107,7 +1116,7 @@ public CompletableFuture<Void> upsert(BinaryRowEx row, @Nullable InternalTransac
                         .transactionId(txo.id())
                         .enlistmentConsistencyToken(enlistmentConsistencyToken)
                         .requestType(RW_UPSERT)
-                        .timestamp(clockService.now())
+                        .timestamp(txo.startTimestamp()) // TODO replace everythere
```

Comment:
We can use startTs here because it doesn't affect transaction ordering guaranties. Proper order is provided by locking, and the response will always have timestamp greater or equal to latest committed dependent transaction. See IEP-91 for more details.

Reply:
I've created https://issues.apache.org/jira/browse/IGNITE-23712 for TODO